### ISSUE TITLE
parameter-overrides broken with json as value

### ISSUE THREAD

**Issue by revdbrink** [awslabs/aws-sam-cli#1602]
Since 0.33.1 it's not possible to to use JSON as value for --parameter-overrides

This worked before:
`sam deploy --template-file output.yaml --stack-name my-stack --capabilities CAPABILITY_NAMED_IAM CAPABILITY_AUTO_EXPAND --parameter-overrides DeployStackName=new-stack  DeployParameterOverrides={"bucketName":"production","bucketRegion":"eu-west-1"} DeployParameterBucketOverrides={"bucketName":"myownbucket"}`

This is now broken by commit https://github.com/awslabs/aws-sam-cli/commit/9d2063ad14a0c8797f39950cb35fbbe8451bcad1

I tried to fix it with(out) other quotes or extra escaping but that makes not different.

**Comment by plantusd** [comment]
I also noticed this, while locally, using Windows 10, I could just use  --parameter-overrides Key=Value, now it (since version 3) changed to ParameterKey=Key,ParameterValue=Value format while it adds consistent, I didn't see it in Release notes

**Comment by TheSriram** [comment]
Single quote the entire thing, if the values have quotes (after escaping) and try if that fixes it.

This is the below output i got.

```
❯ samdev deploy --stack-name try --parameter-overrides 'DeployStackName=new-stack DeployParameterOverrides="{\"bucketName\":\"production\",\"bucketRegion\":\"eu-west-1\"}" DeployParameterBucketOverrides="{\"bucketName\":\"myownbucket\"}"'

	Deploying with following values
	===============================
	Stack name                 : try
	Region                     : us-east-1
	Confirm changeset          : False
	Deployment s3 bucket       : my-sam-app-bucket1
	Capabilities               : ["CAPABILITY_IAM"]
	Parameter overrides        : {'DeployStackName': 'new-stack', 'DeployParameterOverrides': '{"bucketName":"production","bucketRegion":"eu-west-1"}', 'DeployParameterBucketOverrides': '{"bucketName":"myownbucket"}'}
```

**Comment by revdbrink** [comment]
This works for me, tnx!

**Comment by revdbrink** [comment]
The sam deploy works but the value in my stack parameter for DeployParameterOverrides is now only '{' and not the expected '{"bucketName":"production","bucketRegion":"eu-west-1"}'

**Comment by jfuss** [comment]
@revdbrink You need to escape the quotes.

**Comment by revdbrink** [comment]
that solves it! Thanks for the quick response.

**Comment by ryancabanas** [comment]
Hi,

I'm having a similar issue to this that just started today.  I'm using a `samconfig.toml` file with the `parameter_overrides` option which has a parameter in it that needs a JSON value.  I have successfully been using this config file, but all of a sudden, today it has stopped working when I attempt to do a `sam deploy`.

My `parameter_overrides` line looks like this:

`parameter_overrides = "ParameterOverrides=\"{\"Stage\" : \"dev\"}\""`

When SAM shows me what it's going to deploy for the `parameter_overrides`, it keeps showing me a single `{`, like @revdbrink [mentioned](https://github.com/aws/aws-sam-cli/issues/1602#issuecomment-559046208), like this:

`{"ParameterOverrides": "{"}`

I tried single quotes and escaping the double quotes like @jfuss suggested, but that didn't work for me.

I hadn't upgraded my SAM CLI from when it was working before.  I was on 1.23.0.  I tried upgrading to 1.27.2, but it didn't make a difference.

Any thoughts on why this would all of a sudden stop working?  Am I missing something totally obvious?  Thanks.
